library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -1880,6 +1887,58 @@ protected static int parseMpegChannelConfiguration(XmlPullParser xpp) {
         : Format.NO_VALUE;
   }
 
+  /**
+   * Parses the number of channels from the value attribute of an AudioElementConfiguration with
+   * schemeIdUri as defined by Annex G (3.2) in ETSI TS 102 114 V1.6.1
+   *
+   * @param xpp The parser from which to read.
+   * @return The parsed number of channels, or {@link Format#NO_VALUE} if the channel count could
+   * not be parsed.
+   */
+  protected static int parseDtsChannelConfiguration(XmlPullParser xpp) {
+    @Nullable String value = xpp.getAttributeValue(null, "value");
+    if (value == null) {
+      return Format.NO_VALUE;
+    } else {
+      int channelCount = Integer.decode(value);
+      if ((channelCount > 0) && (channelCount < 33)) {
+        return channelCount;
+      } else {
+        return Format.NO_VALUE;
+      }
+    }
+  }
+
+  /**
+   * Parses the number of channels from the value attribute of an AudioElementConfiguration with
+   * schemeIdUri as defined by table B-5 in ETSI TS 103 491 v1.2.1.
+   *
+   * @param xpp The parser from which to read.
+   * @return The parsed number of channels, or {@link Format#NO_VALUE} if the channel count could
+   * not be parsed.
+   */
+  protected static int parseDtsxChannelConfiguration(XmlPullParser xpp) {
```

Comment:
Not sure (so please check), but is it equivalent to do:

```
protected static int parseDtsxChannelConfiguration(XmlPullParser xpp) {
  @Nullable String value = xpp.getAttributeValue(null, "value");
  if (value == null) {
    return Format.NO_VALUE;
  }
  int channelCount = Integer.bitCount(Integer.parseInt(value, /* radix= */ 16));
  return channelCount == 0 ? Format.NO_VALUE : channelCount;
}
```

I guess the main change is to use `Integer.bitCount`, on the assumption that it does the same thing?

Reply:
Yes. That's fine. It is equivalent.  Thanks.